handle backslash in pathname

lxhillwind opened the pull request:
This solves https://github.com/justinmk/vim-dirvish/issues/189.

justinmk commented on syntax/dirvish.vim:
```diff
@@ -3,13 +3,14 @@ if 'dirvish' !=# get(b:, 'current_syntax', 'dirvish')
 endif
 
 let s:sep = exists('+shellslash') && !&shellslash ? '\' : '/'
+let s:sep_escaped = s:sep == '\' ? '\\' : '/'
```
nit: let's stick with `_esc` suffix to be consistent with similar variables in the plugin.
```suggestion
let s:sep_esc = s:sep == '\' ? '\\' : '/'
```

justinmk commented on ftplugin/dirvish.vim:
```diff
@@ -5,6 +5,8 @@ let b:did_ftplugin = 1
 
 let s:nowait = (v:version > 703 ? '<nowait>' : '')
 
+let s:sep = exists('+shellslash') && !&shellslash ? '\' : '/'
```
nit: delete blank line so that the `s:foo` variables are grouped

lxhillwind commented:
I have rebased on current origin/master (b136fd27a0afaee8306655eca7ac19d0e780faa3) and modified according to your suggestion, so that it should be ok to merge.
